### PR TITLE
[IMP] l10n_in: add crores corresponding to digits of HSN in settings

### PR DESCRIPTION
**Pull request by masm-odoo**
Modified HSN display format based on crore thresholds:
  - 4-digit HSN for values below 5 crore.
  - 6-digit HSN for values above 5 crore.

This update enhances clarity by providing distinct formats for HSN codes based on value thresholds.

> Task: 4252228

### CODE REVIEW

**Review by yosuanicolaus — COMMENTED** [review]
you also need to update the `.pot` file for this module :)

File: addons/l10n_in/models/company.py [hunk 1]
```diff
@@ -9,8 +9,8 @@ class ResCompany(models.Model):
     l10n_in_upi_id = fields.Char(string="UPI Id")
     l10n_in_hsn_code_digit = fields.Selection(
         selection=[
-            ("4", "4 Digits"),
-            ("6", "6 Digits"),
+            ("4", "4 Digits (turnover < 5 CR.)"),
+            ("6", "6 Digits (turnover > 5 CR.)"),
```
Comment (hamo-odoo) [inline addons/l10n_in/models/company.py:13]:
```suggestion
            ("4", "4 Digits"),
            ("6", "6 Digits"),
```
IMO this looks a bit ugly on the view though it was my suggestion

**Review by hamo-odoo — CHANGES_REQUESTED** [review]
image
```diff
diff --git a/addons/l10n_in/views/res_config_settings_views.xml b/addons/l10n_in/views/res_config_settings_views.xml
index 166926568b29..2dc82cc89d20 100644
--- a/addons/l10n_in/views/res_config_settings_views.xml
+++ b/addons/l10n_in/views/res_config_settings_views.xml
@@ -56,7 +56,11 @@
             </xpath>
             <app name="account" position="inside">
                 <block title="Product" name="l10n_in_product_setting_container" invisible="country_code != 'IN'">
-                    <setting string="HSN/SAC Validation" help="HSN/SAC Digit Validation for GST Compliance based on your Aggregate Annual Turnover (AATO).">
+                    <setting string="HSN/SAC Validation" help="HSN/SAC Validation for GST Compliance based on your Aggregate Annual Turnover (AATO):" company_dependent="1">
+                    <div class="text-muted">
+                        <p class="my-0 ms-1">- 4 Digits for turnover &lt; ₹5 Crores</p>
+                        <p class="ms-1">- 6 Digits for turnover &gt; ₹5 Crores</p>
+                    </div>
                         <field name="l10n_in_hsn_code_digit"/>
                     </setting>
                 </block>
```
I feel this would be much better for a UX also we are missing the company dependent in the view

@jco-odoo I hope you are okay with this. Better to give user something a way better than something looking more ugly

File: addons/l10n_in/views/res_config_settings_views.xml [hunk 2]
```diff
@@ -56,7 +56,11 @@
             </xpath>
             <app name="account" position="inside">
                 <block title="Product" name="l10n_in_product_setting_container" invisible="country_code != 'IN'">
-                    <setting string="Enable HSN/SAC Validation" help="HSN/SAC Digit Validation for GST Compliance based on your Aggregate Annual Turnover (AATO).">
+                    <setting string="HSN/SAC Validation" help="HSN/SAC Validation for GST Compliance based on your Aggregate Annual Turnover (AATO):" company_dependent="1">
+                        <div class="text-muted">
+                            <p class="my-0 ms-1">• 4 Digits for turnover &lt; ₹5 Crores</p>
+                            <p class="ms-1">• 6 Digits for turnover &gt; ₹5 Crores</p>
```
Comment (hamo-odoo) [inline addons/l10n_in/views/res_config_settings_views.xml:62]:
```suggestion
                            <ul>
                                <li> 4 Digits for turnover &lt; ₹5 Crores</li>
                                <li> 6 Digits for turnover &gt; ₹5 Crores</li>
                            </ul>
```

**Review by hamo-odoo — APPROVED** [review]

File: addons/l10n_in/i18n/l10n_in.pot [hunk 3]
```diff
@@ -273,7 +283,7 @@ msgstr ""
 
 #. module: l10n_in
 #: model_terms:ir.ui.view,arch_db:l10n_in.res_config_settings_view_form_inherit_l10n_in
-msgid "Enable HSN/SAC Validation"
+msgid "HSN/SAC Validation"
 msgstr ""
```
Comment (yosuanicolaus) [inline addons/l10n_in/i18n/l10n_in.pot:288]: This looks clean but then it won't be alphabetically sorted. Any new generated `.pot` files will have this translation down among with the "H" stuffs, and will need to be re-positioned everytime.

I suggest just keep it simple, let these lines be deleted and write the new one in the default place. It's not that much lines of code to add anyway

**Review by yosuanicolaus — CHANGES_REQUESTED** [review]

**Review by aboo-odoo — COMMENTED** [review]
LGTM! @jco-odoo

**Comment by hamo-odoo** [comment]
@jco-odoo Changes done ✅ . Good to go for r+

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**